src/passkey_child/passkey_child_common.c

```diff
@@ -883,3 +904,47 @@ verify_assert_data(struct passkey_data *data)
 
     return ret;
 }
+
+errno_t
+preflight(struct passkey_data *data, int timeout)
+{
+    fido_assert_t *assert = NULL;
+    fido_dev_t *dev = NULL;
+    int index = 0;
+    int pin_retries = 0;
+    errno_t ret;
+
+    ret = select_authenticator(data, timeout, &dev, &assert, &index);
+    if (ret != EOK) {
+        goto done;
+    }
+
+    DEBUG(SSSDBG_TRACE_FUNC, "Comparing the device and policy options.\n");
+    ret = get_device_options(dev, data);
+    if (ret != FIDO_OK) {
+        goto done;
+    }
+
+    DEBUG(SSSDBG_TRACE_FUNC, "Checking the number of remaining PIN retries.\n");
+    ret = get_device_pin_retries(dev, data, &pin_retries);
+    if (ret != FIDO_OK) {
+        goto done;
+    }
+
+    ret = EOK;
+
+done:
+    if (ret != EOK) {
+        data->user_verification = FIDO_OPT_TRUE;
+        pin_retries = MAX_PIN_RETRIES;
```

Comment:
Similar to my other comment about printing default values, why set these to some typical value if `ret != EOK` ? We should error out instead?

Reply:
This has been clarified in my first reply